srv/components/system/config/set_hostname.sls

```diff
@@ -15,9 +15,12 @@
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 #
 
+Update hostname file permissions:
+  cmd.run:
+    - name: "chattr -i /etc/hostname"
+
 Set hostname:
-  network.system:
-    - name: set_hostname
-    - hostname: {{ pillar['cluster'][grains['id']]['hostname'] }}
-    - apply_hostname: True
-    - retain_settings: True
+  cmd.run:
+    - name: "hostnamectl set-hostname {{ pillar['cluster'][grains['id']]['hostname'] }}"
```

Comment:
Prefer converting this to a custom module.
Then handle the `chattr` driven immutability inside the custom set_hostname module.

Reply:
Used `network.system` module instead of set-hostname command